embassy-stm32/src/gpio.rs

```diff
@@ -148,6 +184,18 @@ impl<'d, T: Pin> StatefulOutputPin for Output<'d, T> {
     }
 }
 
+impl<'d, T: Pin> InputPin for Output<'d, T> {
```

Comment:
This is a bit dangerous, if the user is creating an Output they shouldn't be able to use it as input except in very special circumstances. I guess you want this for onewire?

Maybe it should be a new `InputOutput` driver? Maybe even `OpenDrainInputOutput` because it only makes sense for opendrain?